### PR TITLE
[Admin][Product] Access the variants management from product edit page

### LINKED ISSUE #9647: Add shortcut to variants in product edit page
**Describe the proposed solution**
I want to add a shortcut to the variants when you edit a product. At the moment you have to go back to the product list, click manage variants and then list variants.

It's much along the lines of #9565.

screen shot 2018-08-14 at 11 36 31 copy

### PR DESCRIPTION
**Pull request by GSadee**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | 
| Related tickets | replaces #9785, fixes #9647
| License         | MIT

image

### CODE REVIEW

File: src/Sylius/Bundle/UiBundle/Resources/views/Menu/top.html.twig [hunk 1]
```diff
@@ -41,6 +43,26 @@
...
+
+    <div class="ui{% if icon %} labeled icon{% endif %} floating dropdown {% if color %}{{ color }} {% endif %}link button">
+        {% if icon %}<i class="{{ icon }} icon"></i>{% endif %}
+        <span class="text">{{ item.label|trans }}</span>
+        <div class="menu">
+            {% for child in item.children %}
+                {% set icon = child.labelAttribute('icon') %}
+
+                <a href="{{ child.uri }}" class="item">
+                    {% if icon %}<i class="icon {{ icon }}"></i> {% endif %}
```
Comment (Tomanhez) [inline src/Sylius/Bundle/UiBundle/Resources/views/Menu/top.html.twig]: {% if icon %}<i class="icon {{ icon }}"></i><cmd-g>{% endif %}

Redundant blank space

File: src/Sylius/Bundle/UiBundle/Resources/views/Menu/top.html.twig [hunk 2]
```diff
@@ -41,6 +43,26 @@
     </a>
 {% endblock %}
 
+{% block links %}
+    {% set color = item.labelAttribute('color') %}
+    {% set icon = item.labelAttribute('icon') %}
+
+    <div class="ui{% if icon %} labeled icon{% endif %} floating dropdown {% if color %}{{ color }} {% endif %}link button">
```
Comment (Tomanhez) [inline src/Sylius/Bundle/UiBundle/Resources/views/Menu/top.html.twig]:
```suggestion
    <div class="ui{% if icon %} labeled icon {% endif %} floating dropdown {% if color %}{{ color }} {% endif %}link button">
```
Missing space

Comment (GSadee) [reply src/Sylius/Bundle/UiBundle/Resources/views/Menu/top.html.twig]: This space is intentionally missing

File: src/Sylius/Bundle/UiBundle/Resources/views/Menu/top.html.twig (hunk 1 above)
Comment (GSadee) [reply src/Sylius/Bundle/UiBundle/Resources/views/Menu/top.html.twig]: This space is intentionally missing

Comment (CoderMaggie) [reply src/Sylius/Bundle/UiBundle/Resources/views/Menu/top.html.twig]: This space is needed to have space between label and icon I suppose. ❓

File: src/Sylius/Bundle/AdminBundle/Menu/ProductUpdateMenuBuilder.php [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    public function createMenu(array $options): ItemInterface
+    {
+        $menu = $this->factory->createItem('root');
+
+        if (!isset($options['product'])) {
+            return $menu;
+        }
+
+        $product = $options['product'];
+        if (!$product instanceof ProductInterface) {
```
Comment (lchrusciel) [inline src/Sylius/Bundle/AdminBundle/Menu/ProductUpdateMenuBuilder.php]: Can we also skip this logic for simple products?

File: src/Sylius/Bundle/UiBundle/Resources/views/Menu/top.html.twig (hunk 2 above)
Comment (Tomanhez) [inline src/Sylius/Bundle/UiBundle/Resources/views/Menu/top.html.twig]:
```suggestion
    <div class="ui{% if icon %} labeled icon {% endif %} floating dropdown {% if color %}{{ color }} {% endif %}link button">
```

missing space

File: src/Sylius/Bundle/AdminBundle/Menu/ProductUpdateMenuBuilder.php (hunk 3 above)
Comment (GSadee) [reply src/Sylius/Bundle/AdminBundle/Menu/ProductUpdateMenuBuilder.php]: For simple products, there is possibility to create variant (the same behaviour as on index page)

**Comment by GSadee** [comment]
@lchrusciel I've updated docs 👌

**Comment by Zales0123** [comment]
Thank you, Grzegorz! :tada:

**Pull request merged**